devito/types/sparse.py

```diff
@@ -469,30 +467,29 @@ class SparseFunction(AbstractSparseFunction):
 
     _sub_functions = ('coordinates',)
 
-    def __init__(self, *args, **kwargs):
-        if not self._cached():
-            super(SparseFunction, self).__init__(*args, **kwargs)
+    def __init_finalize__(self, *args, **kwargs):
```

Comment:
note for reviewers: this is exactly like before, but de-indented as we don't need ``if not self._cached()`` anymore